src/main/java/edu/oregonstate/cs361/battleship/ComputerAI.java

```diff
@@ -1,66 +1,235 @@
 
-        cpuTargetFired.setRow(cpuFireRow);
-        cpuTargetFired.setCol(cpuFireCol);
-        cpuFireCol += 2;
-        if (cpuFireCol > numCol) {
-            cpuFireRow += 1;
-            cpuFireCol -= numCol;
-            if (numCol % 2 == 0) {
-                if (cpuFireRow % 2 == 0){
-                    cpuFireCol += 1;
+        for(int i = 0; i < outOfBounds.size(); i++) {
+            if(dirtyAdjCoords.contains(outOfBounds.get(i))) {
+                dirtyAdjCoords.remove(dirtyAdjCoords.indexOf(outOfBounds.get(i)));
+            }
+        }
+
+        for(int i = 0; i < misses.size(); i++) {
+            if(dirtyAdjCoords.contains(misses.get(i))) {
+                dirtyAdjCoords.remove(dirtyAdjCoords.indexOf(misses.get(i)));
+            }
+        }
+
+        for(int i = 0; i < hits.size(); i++) {
+            if(dirtyAdjCoords.contains(hits.get(i))) {
+                dirtyAdjCoords.remove(dirtyAdjCoords.indexOf(hits.get(i)));
+            }
+        }
+
+        return dirtyAdjCoords;
+    }
+
+    private ArrayList<Point> hitLiveShipChecker(BattleshipModel cpuModel) {
+        ArrayList<Ship> deadShips = cpuModel.getDestroyedPlayerShips();
+        ArrayList<Point> hits = cpuModel.getComputerHits();
+        Point temp;
+
+        for(int i = 0; i < deadShips.size(); i++) {
+            for(int j = 0; j < deadShips.get(i).getLength(); j++) {
+                temp = deadShips.get(i).getStart();
+                //if current dead ship is horizontal
+                if(deadShips.get(i).getStart().getRow() == deadShips.get(i).getEnd().getRow()) {
+                    temp.setCol(temp.getCol() + j);
+
+                    if(hits.contains(temp)) {
+                        hits.remove(hits.indexOf(temp));
+                    }
                 }
+
+                //else current dead ship is vertical
                 else {
-                    cpuFireCol -= 1;
+                    temp.setRow(temp.getRow() + j);
+
+                    if(hits.contains(temp)) {
+                        hits.remove(hits.indexOf(temp));
+                    }
                 }
             }
         }
-        if (cpuFireRow > numRow) {
-            cpuFireRow -= numRow;
+
+        return hits;
+    }
+
+    private boolean fireOverlapChecker(BattleshipModel cpuModel, int row, int col) {
+        ArrayList<Point> cpuHits = cpuModel.getComputerHits();
+        ArrayList<Point> cpuMisses = cpuModel.getComputermisses();
+
+        for(int i = 0; i < cpuHits.size(); i++) {
+            if((row == cpuHits.get(i).getRow()) && (col == cpuHits.get(i).getCol())) {
+                return true;
+            }
         }
-        return cpuTargetFired;
+
+        for(int i = 0; i < cpuMisses.size(); i++) {
+            if((row == cpuMisses.get(i).getRow()) && (col == cpuMisses.get(i).getCol())) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
+        //Currently this AI will loop from left-top to every other space.
```

Comment:
Maybe we can remove some of these larger comment blocks before the final submission, so as to improve the readability of the code.